google-apis-generator/lib/google/apis/generator/templates/_method.tmpl

```diff
@@ -50,7 +50,15 @@
 # @raise [Google::Apis::ClientError] The request is invalid and should not be retried without modification
 # @raise [Google::Apis::AuthorizationError] Authorization is required
 def <%= api_method.generated_name %>(<% for param in api_method.required_parameters %><%= param.generated_name %>, <% end %><% if api_method.request %><%= api.schemas[api_method.request._ref].generated_name %>_object = nil, <% end %><% for param in api_method.optional_query_parameters %><%= param.generated_name %>: nil, <% end %><% for param in api.parameters.values.reject {|p| p.name == 'key'} %><%= param.generated_name %>: nil, <% end %><% if api_method.supports_media_upload? %>upload_source: nil, content_type: nil, <% elsif api_method.supports_media_download? %>download_dest: nil, <%  end  %>options: nil, &block)
-<% if api_method.supports_media_upload? -%>
+<% if api_method.supports_media_upload? && api.name == 'Storage' %>
+  if upload_source.nil?
+    command = make_simple_command(:<%= api_method.http_method.downcase %>, '<%= api_method.path %>', options)
+  else
+    command = make_storage_upload_command(:<%= api_method.http_method.downcase %>, '<%= api_method.path %>', options)
```

Comment:
Does this need to be changed in the generated code in this PR also? In Go, presubmits will fail if the generated code doesn't match.

Reply:
Once this change is merged, the generator will generate the new code and raise a PR. Or I'll do it manually.